auslib/admin/views/permissions.py

```diff
@@ -95,3 +95,41 @@ def _delete(self, username, permission, changed_by, transaction):
         except ValueError as e:
             self.log.warning("Bad input: %s", e.args)
             return Response(status=400, response=e.args)
+
+
+class UserRolesView(AdminView):
+    """/users/:username/roles"""
+
+    def get(self, username):
+        roles = dbo.permissions.user_roles.getRoles(username)
+        if roles:
+            return jsonify({"roles": roles})
+        else:
+            return Response(status=404, response="No roles found for user")
+
+
+class UserRoleView(AdminView):
+    """/users/:username/roles/:role"""
+
+    @requirelogin
+    def _put(self, username, role, changed_by, transaction):
+        r = dbo.permissions.user_roles.select({"username": username, "role": role}, transaction=transaction)
+        if r:
+            return Response(status=200, response=json.dumps({"new_data_version": r[0]["data_version"]}))
```

Comment:
Could you please add a comment to the effect of 'adding a role which already exists is treated as a no-op'.